code/datums/components/forensics.dm

```diff
@@ -0,0 +1,165 @@
+/datum/component/forensics
+	var/list/fingerprints		//assoc print = print
+	var/list/hiddenprints		//assoc ckey = realname/gloves/ckey
+	var/list/blood_DNA			//assoc dna = bloodtype
+	var/list/fibers				//assoc print = print
+
+/datum/component/forensics/Initialize()
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+	RegisterSignal(COMSIG_COMPONENT_CLEAN_ACT, .proc/clean_act)
+
+/datum/component/forensics/proc/return_fingerprints()
+	return length(fingerprints)? fingerprints.Copy() : list()
+
+/datum/component/forensics/proc/return_hiddenprints()
+	return length(hiddenprints)? hiddenprints.Copy() : list()
+
+/datum/component/forensics/proc/return_blood_DNA()
+	return length(blood_DNA)? blood_DNA.Copy() : list()
+
+/datum/component/forensics/proc/return_fibers()
+	return length(fibers)? fibers.Copy() : list()
+
+/datum/component/forensics/proc/has_fingerprints()
+	return length(fingerprints)? TRUE : FALSE
+
+/datum/component/forensics/proc/has_hiddenprints()
+	return length(hiddenprints)? TRUE : FALSE
+
+/datum/component/forensics/proc/has_blood_DNA()
+	return length(blood_DNA)? TRUE : FALSE
+
+/datum/component/forensics/proc/has_fibers()
+	return length(fibers)? TRUE : FALSE

```

Comment:
A lovely set of examples of how to not do procs

Reply:
could use a define though

Reply:
HAS_FIBRES length(fibres)